File: apps/docs/src/code/python/basics/process_stream_stderr.py
from e2b import Sandbox

sandbox = Sandbox(
    template="base",
    on_stderr=lambda output: print("[sandbox]", output.line),  # $HighlightLine
)

# This command will fail and output to stderr because Golang isn't installed in the cloud playground
proc = sandbox.process.start("go version")
proc.wait()
# output: [sandbox] /bin/bash: line 1: go: command not found

proc_with_custom_handler = sandbox.process.start(
    "go version",
    on_stderr=lambda output: print("[process]", output.line),  # $HighlightLine
)
proc_with_custom_handler.wait()
# output: [process] /bin/bash: line 1: go: command not found

sandbox.close()

File: apps/docs/src/code/js/basics/process_stream_stderr.js
import { Sandbox } from '@e2b/sdk'

const sandbox = await Sandbox.create({
  id: 'base',
})

// This command will fail and output to stderr because Golang isn't installed in the cloud playground
const golangVersion = await sandbox.process.start({
  cmd: 'go version',
  onStderr: (output) => console.log('sandbox', output.line), // $HighlightLine
})
await golangVersion.wait()
// output: [sandbox] /bin/bash: line 1: go: command not found

const procWithCustomHandler = await sandbox.process.start({
  cmd: 'go version',
  onStderr: (data) => console.log('process', data.line), // $HighlightLine
})
await procWithCustomHandler.wait()
// output: process /bin/bash: line 1: go: command not found

await sandbox.close()
